### PR TITLE
Mark instruction helpers appropriately as constexpr

### PR DESCRIPTION
**Pull request by axic**
Pulled out of #562.

### CODE REVIEW

File: lib/fizzy/execute.cpp [hunk 1]
```diff
@@ -423,7 +423,7 @@ __attribute__((no_sanitize("float-divide-by-zero"))) inline constexpr T fdiv(T a
 }
 
 template <typename T>
-inline constexpr T fmin(T a, T b) noexcept
+inline T fmin(T a, T b) noexcept
```
Comment (axic) [inline lib/fizzy/execute.cpp:426]: Both `fmin` and `fmax` use `std::signbit` which is not constexpr: https://en.cppreference.com/w/cpp/numeric/math/signbit

**Comment by axic** [comment]
@chfast could you benchmark this?

**Comment by axic** [comment]
I don't see any change in any of the functions touched by this. Some stuff is moved around, so can't tell much with benchmarking.

**Comment by chfast** [comment]
> I don't see any change in any of the functions touched by this. Some stuff is moved around, so can't tell much with benchmarking.

The `execute.cpp.s` is identical in both cases. As it should be.

**Review by chfast — APPROVED** [review]

**Comment by axic** [comment]
> > I don't see any change in any of the functions touched by this. Some stuff is moved around, so can't tell much with benchmarking.
>
> The execute.cpp.s is identical in both cases. As it should be.

It wasn't identical for me. Let me check again.

**Comment by axic** [comment]
> > > I don't see any change in any of the functions touched by this. Some stuff is moved around, so can't tell much with benchmarking.
> >
> > The execute.cpp.s is identical in both cases. As it should be.
>
> It wasn't identical for me. Let me check again.
>
> Still not identical, interestingly the resulting .S is the same number of characters, but the content is different.

I totally forgot to consider optimiser settings when comparing the output. With the settings of the release build indeed it is identical.

**Pull request merged**